misaligned.c

```diff
@@ -1,21 +1,42 @@
 #include <stdio.h>
 #include <assert.h>
+#include "misaligned.h"
 
 int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+            printf("%d , %s , %s\n", i * 5 + j, majorColorNames[i], minorColorNames[j]);
         }
     }
     return i * j;
 }
 
+ColorPair GetColorFromPairNumber(int pairNumber) {
+    ColorPair colorPair;
+    int zeroBasedPairNumber = pairNumber - 1;
+    colorPair.majorColorValue = 
+        (enum MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+    colorPair.minorColorValue =
+        (enum MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+    return colorPair;
+}
+
+void testNumberToPair(int pairNumber,
+    enum MajorColor expectedMajor,
+    enum MinorColor expectedMinor)
+{
+    ColorPair colorPair = GetColorFromPairNumber(pairNumber);
+    assert(colorPair.minorColorValue == expectedMinor);
+    assert(colorPair.majorColorValue == expectedMajor);
+}
+
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
+    printColorMap();

```

Comment:
How is the alignment of output strings verified? test case needs to be added for alignment verification.
Also note that there are single and double digit pair numbers, and variable length color names.